src/Components/Web.JS/src/Rendering/EventForDotNet.ts

```diff
@@ -204,8 +211,38 @@ function parseMouseEvent(event: MouseEvent) {
   };
 }
 
-function isCheckbox(element: Element | null) {
-  return element && element.tagName === 'INPUT' && element.getAttribute('type') === 'checkbox';
+function isCheckbox(element: Element | null): boolean {
+  return !!element && element.tagName === 'INPUT' && element.getAttribute('type') === 'checkbox';
+}
+
+const timeBasedInputs = [
+  'date',
+  'datetime-local',
+  'month',
+  'time',
+  'week',
+];
+
+function isTimeBasedInput(element: Element | null): boolean {
+  return !!element && timeBasedInputs.indexOf(element.getAttribute('type') || '') !== -1;
```

Comment:
I think this can be simplified - it looks like the type info already tells us that `element` is nonnull at the caller, so this function could be reduced to:

```ts
function isTimeBasedInput(element: Element): boolean {
  return timeBasedInputs.indexOf(element.getAttribute('type')) !== -1;
}
```

Reply:
Also, not sure if you like doing fancy stuff in TypeScript, but you could do this:

```ts
function isTimeBasedInput(element: Element): element is HTMLInputElement {
    return timeBasedInputs.indexOf(element.getAttribute('type')!) !== -1;
}
```

... and then `normalizeTimeBasedValue` could be declared as:

```ts
function normalizeTimeBasedValue(element: HTMLInputElement): string { ... }
```

... and then it could access `element.type` and `element.value` directly, not needing `element['type']` or `element['value']`.